### PR TITLE
PT-4002 (stable): Reindexing HPO fails when an HPO term has more than one single value type field

### PR DESCRIPTION
**Pull request by veronikaslc**
Fixed.

### CODE REVIEW

File: components/vocabularies/api/src/main/java/org/phenotips/vocabulary/internal/solr/AbstractOBOSolrVocabulary.java [hunk 1]
```diff
@@ -118,12 +128,23 @@ protected int index(String sourceUrl)
                     termBatch = new HashSet<>();
                     batchCounter = 0;
                 }
+                singleValuedFileds.forEach(fieldName -> singleValueFieldsTracker.put(fieldName, false));
```
Comment (allasm) [inline components/vocabularies/api/src/main/java/org/phenotips/vocabulary/internal/solr/AbstractOBOSolrVocabulary.java]: you only need `singleValueFieldsTracker` here, so the right way is to define it here, and initialize it here.

You also don't need a `Map` in that case, it can be done much simpler, as a `Set`, where a presense of  afield indicates it has been encountered already, no need to add a `true` to that

E.g. on this line initialize `Set<String> alreadyPresentFields = new HashSet<>();`

then add all fields encountered on line #146 (can even remove an `if` to reduce cyclomatic complexity).

this way also never have to clear the Set

File: components/vocabularies/chebi/api/src/test/java/org/phenotips/vocabulary/internal/solr/ChEBIOntologyTest.java [hunk 2]
```diff
@@ -88,7 +88,6 @@ public void testChEBIOntologyReindex()
         Mockito.verify(this.server, Mockito.atLeast(1)).commit();
         Mockito.verify(this.server, Mockito.atLeast(1)).add(Matchers.anyCollectionOf(SolrInputDocument.class));
         Mockito.verify(this.cache, Mockito.atLeast(1)).removeAll();
-        Mockito.verifyNoMoreInteractions(this.cache, this.server);
```
Comment (allasm) [inline components/vocabularies/chebi/api/src/test/java/org/phenotips/vocabulary/internal/solr/ChEBIOntologyTest.java]: why this change btw?

Comment (veronikaslc) [reply components/vocabularies/chebi/api/src/test/java/org/phenotips/vocabulary/internal/solr/ChEBIOntologyTest.java]: because of the interaction with SolrClient to read single-valued schema fields.

Comment (allasm) [reply components/vocabularies/chebi/api/src/test/java/org/phenotips/vocabulary/internal/solr/ChEBIOntologyTest.java]: I think there is a way to say "exactly one interaction"
